x/stakeibc/keeper/msg_server_liquid_stake.go

```diff
@@ -17,114 +15,91 @@ import (
 	rateIsSafe, err := k.IsRedemptionRateWithinSafetyBounds(ctx, *hostZone)
 	if !rateIsSafe || (err != nil) {
-		errMsg := fmt.Sprintf("IsRedemptionRateWithinSafetyBounds check failed. hostZone: %s, err: %s", hostZone.String(), err.Error())
-		return nil, errorsmod.Wrapf(types.ErrRedemptionRateOutsideSafetyBounds, errMsg)
+		return nil, errorsmod.Wrapf(types.ErrRedemptionRateOutsideSafetyBounds, "HostZone: %s, err: %s", hostZone.ChainId, err.Error())
 	}
 
-	bech32ZoneAddress, err := sdk.AccAddressFromBech32(hostZone.Address)
-	if err != nil {
-		return nil, fmt.Errorf("could not bech32 decode address %s of zone with id: %s", hostZone.Address, hostZone.ChainId)
-	}
-	err = k.bankKeeper.SendCoins(ctx, sender, bech32ZoneAddress, sdk.NewCoins(inCoin))
-	if err != nil {
-		k.Logger(ctx).Error("failed to send tokens from Account to Module")
-		return nil, errorsmod.Wrap(err, "failed to send tokens from Account to Module")
-	}
-	// mint user `amount` of the corresponding stAsset
-	// NOTE: We should ensure that denoms are unique - we don't want anyone spoofing denoms
-	err = k.MintStAssetAndTransfer(ctx, sender, msg.Amount, msg.HostDenom)
-	if err != nil {
-		k.Logger(ctx).Error("failed to send tokens from Account to Module")
-		return nil, errorsmod.Wrapf(err, "failed to mint %s stAssets to user", msg.HostDenom)
-	}
-
-	// create a deposit record of these tokens (pending transfer)
+	// Grab the deposit record that will be used for record keeping
 	strideEpochTracker, found := k.GetEpochTracker(ctx, epochtypes.STRIDE_EPOCH)
 	if !found {
-		k.Logger(ctx).Error("failed to find stride epoch")
 		return nil, errorsmod.Wrapf(sdkerrors.ErrNotFound, "no epoch number for epoch (%s)", epochtypes.STRIDE_EPOCH)
 	}
-	// Does this use too much gas?
-	depositRecord, found := k.RecordsKeeper.GetDepositRecordByEpochAndChain(ctx, strideEpochTracker.EpochNumber, hostZone.ChainId)
+	depositRecord, found := k.RecordsKeeper.GetTransferDepositRecordByEpochAndChain(ctx, strideEpochTracker.EpochNumber, hostZone.ChainId)
 	if !found {
-		k.Logger(ctx).Error("failed to find deposit record")
-		return nil, errorsmod.Wrapf(sdkerrors.ErrNotFound, fmt.Sprintf("no deposit record for epoch (%d)", strideEpochTracker.EpochNumber))
+		return nil, errorsmod.Wrapf(sdkerrors.ErrNotFound, "no deposit record for epoch (%d)", strideEpochTracker.EpochNumber)
 	}
-	depositRecord.Amount = depositRecord.Amount.Add(msg.Amount)
-	k.RecordsKeeper.SetDepositRecord(ctx, *depositRecord)
 
-	k.hooks.AfterLiquidStake(ctx, sender)
-	return &types.MsgLiquidStakeResponse{}, nil
-}
-
-func (k msgServer) MintStAssetAndTransfer(ctx sdk.Context, sender sdk.AccAddress, amount sdkmath.Int, denom string) error {
-	stAssetDenom := types.StAssetDenomFromHostZoneDenom(denom)
+	// The tokens that are sent to the protocol are denominated in the ibc hash of the native token on stride (e.g. ibc/xxx)
+	nativeDenom := hostZone.IbcDenom
+	nativeCoin := sdk.NewCoin(nativeDenom, msg.Amount)
```

Comment:
piconit, should line 52 be below the if-statement?

Reply:
I can move it down! It's more just a question of readability (NewCoin will not fail if the denom is not an ibc denom)